core/kernel/ubsan.c

```diff
@@ -196,11 +196,11 @@ void __ubsan_handle_out_of_bounds(void *data_, void *idx __unused)
 		panic();
 }
 
-void __ubsan_handle_unreachable(struct unreachable_data *data)
+void __ubsan_handle_builtin_unreachable(void *data_)
 {
+	struct unreachable_data *data = data_;
 	print_loc(__func__, &data->loc);
-	if (ubsan_panic)
-		panic();
+	panic();
```

Comment:
Doesn't this result in a warning?

Reply:
Hi,

I think the opposite results in a warning, since this function marked with attribute noreturn. So if it doesn't call explicitly any other noreturn function or doesn't end with infinite loop I get the warning.

```
core/kernel/ubsan.c: In function ‘__ubsan_handle_builtin_unreachable’:
core/kernel/ubsan.c:188:1: warning: ‘noreturn’ function does return
  188 | }
```
That is the reason why I treated noreturn functions in a little bit special way here.

Reply:
That means that this function has an implicit `noreturn` attribute. How about adding it explicitly to be clear?

Reply:
> That means that this function has an implicit noreturn attribute. How about adding it explicitly to be clear?

That was exactly my thought and I tried to do this. But for some weird reason I get a warning message in this case

```
core/kernel/ubsan.c:182:1: warning: ignoring attribute ‘noreturn’ in declaration of a built-in function ‘__ubsan_handle_builtin_unreachable’ because it conflicts with attribute ‘const’ [-Wattributes]
  182 | {
      | ^
```
After some research I found [this](https://patches.linaro.org/project/lkml/patch/20180202154813.1625742-1-arnd@arndb.de/).  So I can mark it as noreturn, but we will see this warning. If you are okay, then I will do that.

Reply:
Fair enough, it seems like a forever bug in GCC, let's keep it as it is.